### PR TITLE
Authorize flow and basic IPN handling

### PR DESCRIPTION
**Pull request by msaggiorato**
This allows authorization flows to happen through IPNs and cron polling mechanisms.

### CODE REVIEW

File: includes/class-wc-gateway-amazon-payments-advanced.php [hunk 1]
```diff
@@ -790,6 +809,60 @@ public function handle_return() {
...
+				$order->update_status( 'on-hold' );
+				wc_maybe_reduce_stock_levels( $order->get_id() );
+				wc_apa()->ipn_handler->schedule_hook( $order->get_id() );
+				break;
+			case 'Canceled':
+			case 'Declined':
+				$order->update_status( 'failed' );
+				wc_maybe_increase_stock_levels( $order->get_id() );
+				break;
+			case 'Captured':
```
Comment (lauramelos) [inline includes/class-wc-gateway-amazon-payments-advanced.php:853]: this event do not change order status?

Comment (lauramelos) [reply includes/class-wc-gateway-amazon-payments-advanced.php:853]: and also consider to add filters to those order status (from a ticket created) Order status shouldn't affect plugin flow and some users need to have different ones on their systems.

**Review by lauramelos — COMMENTED** [review]
Is Ok, I just left a suggestion.

File: includes/class-wc-gateway-amazon-payments-advanced.php (hunk 1 above)
Comment (msaggiorato) [reply includes/class-wc-gateway-amazon-payments-advanced.php:853]: > this event do not change order status?

Yes it does, payment_complete will do that.

> and also consider to add filters to those order status......

Will implement later alongside the V1 fix (if it is to be added to v1 aswell)

**Pull request merged**